torch/library.py

```diff
@@ -827,6 +829,54 @@ def register_kernel(
     return _impl(op, device_types, func, lib=lib, disable_dynamo=True)
 
 
+def register_autocast(
+    op: _op_identifier,
+    device_type: str,
+    cast_inputs: Optional[_dtype] = None,
```

Comment:
Let's make this required. The None case only really makes sense for autograd.Function (the docs say in the None case we run the inside of the autograd.Function.forward with the autocast context, but this is a layering violation for custom operators)